pkg/stub/handler.go

```diff
@@ -98,6 +108,13 @@ func (h *Handler) syncIngressUpdate(ci *ingressv1alpha1.ClusterIngress) error {
 			if err != nil {
 				return fmt.Errorf("couldn't build service: %v", err)
 			}
+			dsRef, err := getDaemonSetOwnerRef(ds)
+			// Don't create the service unless the DaemonSet is ready
+			if err != nil {
+				return fmt.Errorf("failed to create service, could not get DaemonSet ownerReference: %s", err)
+			} else {
```

Comment:
else is redundant, we don't follow this convention, just call service.SetownerReferences() after error handling.